src/common/ArticlePageHeader.ts

```diff
@@ -0,0 +1,12 @@
+import type BreadcrumbProps from "./Breadcrumb"
+
+export interface ArticlePageHeaderProps {
+  breadcrumb: BreadcrumbProps
+  category: string
+  title: string
+  date: string
+  summary: string[]
+  LinkComponent?: any
```

Comment:
hmm why any? :( 
minimally should be a jsx.element/string?

Reply:
Haven't got the time to figure out the appropriate typing for this, so kept it as any for now.

In practice this `LinkComponent` is basically the `Link` from the `next/link` package, but because this library is framework-agnostic, we don't want to import/include this dependency. For the typing to be strict here, we must make sure that it is compatible with at least what `Link` is without including the actual Next.js dependency.

Reply:
hmm i dont understand this. if at the end of the day, the render engine expects certain props (that the next package defines for the component link) and renders the next/link as a component form the next package, arent we already not dependent? 
we could always just use the same shape tho right? even if another framework is used, as long as the shape can be converted it is ok?

Reply:
Hmm no, we are not dependent on the next package, because it is valid for a user to provide the `a` string as the LinkComponent and the component will still render correctly. If we add typings here, we run the risk that the site cannot build when the `next/link` package is actually used. We will need to spend time investigating and experimenting to know what is the appropriate type to use here.

Reply:
discussed offline, we are doing this so that we remain type agnostic